src/main/java/com/hotels/beeju/extensions/ThriftHiveMetaStoreJUnitExtension.java

```
@@ -0,0 +1,93 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.extensions;
+
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hotels.beeju.core.ThriftHiveMetaStoreCore;
+
+/**
+ * A JUnit Extension that creates a Hive Metastore Thrift service backed by a Hive Metastore using an HSQLDB in-memory
+ * database.
+ * <p>
+ * A fresh database instance will be created for each test method.
+ * </p>
+ */
+public class ThriftHiveMetaStoreJUnitExtension extends HiveMetaStoreJUnitExtension {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ThriftHiveMetaStoreJUnitExtension.class);
+  private ThriftHiveMetaStoreCore thriftHiveMetaStoreCore;
+
+  /**
+   * Create a Thrift Hive Metastore service with a pre-created database "test_database".
+   */
+  public ThriftHiveMetaStoreJUnitExtension() {
+    this("test_database");
+  }
+
+  /**
+   * Create a Thrift Hive Metastore service with a pre-created database using the provided name.
+   *
+   * @param databaseName Database name.
+   */
+  public ThriftHiveMetaStoreJUnitExtension(String databaseName){
+    this(databaseName, null);
+  }
+
+  /**
+   * Create a Thrift Hive Metastore service with a pre-created database using the provided name and configuration.
+   *
+   * @param databaseName Database name.
+   * @param configuration Hive configuration properties.
+   */
+  public ThriftHiveMetaStoreJUnitExtension(String databaseName, Map<String, String> configuration){
+    super(databaseName,configuration);
+    thriftHiveMetaStoreCore = new ThriftHiveMetaStoreCore(core);
+  }
+
+  @Override
+  public void beforeTest() {
+    try {
+      thriftHiveMetaStoreCore.initialise();
+      super.beforeTest();
+    } catch (Exception e) {
+      LOG.error("Unable to initialise ThriftHiveMetaStore extension", e);
```

same remove try/catch just let them bubble up